src/node_api.cc

```diff
@@ -2275,7 +2339,9 @@ napi_status napi_instanceof(napi_env env,
   CHECK_TO_OBJECT(env, context, ctor, constructor);
 
   if (!ctor->IsFunction()) {
-    napi_throw_type_error(env, "constructor must be a function");
+    napi_throw_type_error(env,
+                          "ERR_NAPI_CONS_FUNCTION",
+                          "TypeError [Constructor must be a function]");
```

Comment:
I think it gets a bit tricky here. If I read this correctly, the resulting error will look something like `TypeError: TypeError [Constructor must be a function]` when stringified; if this were an internal error thrown from JS, it should look something like `TypeError[ERR_NAPI_CONS_FUNCTION]: Constructor must be a function`.

Node’s internal errors do getter magic that would not be trivial to add here, but can I suggest that to make them look more alike, `set_error_code` also sets `err.name` to `(Type|Range)Error[$code]` in addition to setting `err.code`?